torch/csrc/autograd/FunctionsManual.cpp

```diff
@@ -636,13 +625,57 @@ Tensor repeat_backward(Tensor grad, IntArrayRef repeats, IntArrayRef input_shape
   for (int64_t i = 0; i < num_unsqueezed; ++i) {
     grad = grad.sum(0, false);
   }
-  for (size_t j = num_unsqueezed; j < repeats.size(); ++j) {
+
+  const auto & grad_sizes = grad.sizes();
+  std::vector<int64_t> grad_sizes_(input_dims);
+  std::copy(grad_sizes.begin(), grad_sizes.end(), grad_sizes_.begin());
+  auto grad_iter_ = grad_sizes_.begin();
+  for (size_t j = num_unsqueezed; j < repeats.size(); ++j, ++grad_iter_) {
```

Comment:
(no action required)

I think we can call (reshape + sum) a single time instead multiple times. This may be more efficient (because it fuses operations together, especially the sum) but I'm not completely sure. Here's the idea:

Let the input sizes be `input_dimsize`, the number of repeats be `repeats`, and assume that the grad has the same dimensionality as the input.

Then: we could just reshape the grad in one call to be:
```
grad_reshaped = grad.reshape([ repeats[0], input_dimsize[0] / repeats[0], repeats[1], input_dimsize[1] / repeats[1], repeats[2], input_dimsize[2] / repeats[2], ...])
``` 
And reduce over all of the repeat dimensions with:
```
grad_reshape.sum([0, 2, 4, 6, 8, ...])
```

That being said, the code written as-is already delivers a nice improvement so I think it's fine if we ship it in the current form and worry about optimizing it more in the future

Reply:
I actually was also thinking this implementation. But, in the other hand, we need to do `reshape` and `sum` on several dimension where `repeat == 1`. I should also take them into my test cases to compare the performance between two implementations.

Reply:
> But, in the other hand, we need to do reshape and sum on several dimension where repeat == 1

Yeah, that's the downside so it is hard to gauge whether or not it would actually be faster. One would expect a sum on a dimension of size 1 to be a no-op (or close to one) and but the reshape might cause a copy to happen when it didn't happen before (I'm not sure about this part, would need to think it through with some examples)

Reply:
If you're interested in testing the performance between these two implementations it would be good to try some non-contiguous tensors as well